aeron-test-support/src/main/java/io/aeron/test/driver/RedirectingNameResolver.java

```diff
@@ -0,0 +1,163 @@
+
+public class RedirectingNameResolver implements NameResolver
+{
+    public static final int DISABLE_RESOLUTION = -1;
+    public static final int USE_INITIAL_RESOLUTION_HOST = 0;
+    public static final int USE_RE_RESOLUTION_HOST = 1;
+    public static final int NAME_ENTRY_COUNTER_TYPE_ID = 2001;
+
+    private final Map<String, NameEntry> nameToEntryMap = new Object2ObjectHashMap<>();
+    private final String csvConfiguration;
+
+    public RedirectingNameResolver(final String csvConfiguration)
+    {
+        this.csvConfiguration = csvConfiguration;
+        final String[] lines = csvConfiguration.split("\\|");
+        for (final String line : lines)
+        {
+            final String[] params = line.split(",");
+            if (3 != params.length)
+            {
+                throw new IllegalArgumentException("Expect 3 elements per row");
+            }
+
+            final NameEntry nameEntry = new NameEntry(params[0], params[1], params[2]);
+            nameToEntryMap.put(nameEntry.name, nameEntry);
+        }
+    }
+
+    public void init(final MediaDriver.Context context)
+    {
+        final CountersManager countersManager = context.countersManager();
+
+        for (NameEntry nameEntry : nameToEntryMap.values())
+        {
+            final AtomicCounter atomicCounter = countersManager.newCounter(
+                nameEntry.toString(),
+                NAME_ENTRY_COUNTER_TYPE_ID,
+                mutableDirectBuffer -> mutableDirectBuffer.putStringAscii(0, nameEntry.name));
+            nameEntry.counter(atomicCounter);
+        }
+    }
+
+    public InetAddress resolve(final String name, final String uriParamName, final boolean isReResolution)
+    {
+        final NameEntry nameEntry = nameToEntryMap.get(name);
+        final String hostname = null != nameEntry ? nameEntry.redirectHost(name) : name;
+        return hostname != null ? DefaultNameResolver.INSTANCE.resolve(hostname, uriParamName, isReResolution) : null;
+    }
+
+    public String csvConfiguration()
+    {
+        return csvConfiguration;
+    }
+
+    public static boolean updateNameResolutionStatus(
+        final CountersReader counters,
+        final String hostname,
+        final int operationValue)
+    {
+        final MutableInteger nameCounterId = new MutableInteger(NULL_VALUE);
+        counters.forEach((counterId, typeId, keyBuffer, label) ->
+        {
+            if (typeId == NAME_ENTRY_COUNTER_TYPE_ID && hostname.equals(keyBuffer.getStringAscii(0)))
+            {
+                nameCounterId.set(counterId);
+            }
+        });
+
+        final boolean counterFound = NULL_VALUE != nameCounterId.get();
+        if (counterFound)
+        {
+            final AtomicCounter nameCounter = new AtomicCounter(
+                counters.valuesBuffer(),
+                nameCounterId.get());
+
+            nameCounter.set(operationValue);
+        }
+
+        return counterFound;
+    }
+
+    private static final class NameEntry
```

Comment:
Best not to use private classes as this will result in the generation of bridge classes < Java 11.

Reply:
Should we use package-private inner classes or declare them as a separate unit?

Reply:
Package private is a good way to go.